### PR TITLE
add api query getFungibleAssetActivities

### PR DESCRIPTION
**Pull request by 0xjinn**
### Description

Add getFungibleAssetActivities api query

### Test Plan

pnpm test
pnpm fmt

### CODE REVIEW

**Comment by 0xjinn** [comment]
Current dependencies on/for this PR:
* main
  * **PR #73** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-ts-sdk/73" target="_blank">Graphite</a> 
    * **PR #78** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-ts-sdk/78" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-ts-sdk/78?utm_source=stack-comment).

File: src/api/fungible_asset.ts [hunk 1]
```diff
@@ -36,4 +41,26 @@ export class FungibleAsset {
...
+   * It can be filtered by owner address, asset type, and type.
+   *
+   * @param options.pagination Where offset and limit can be specified
+   * @param options.whereCondition Where condition to filter the results
+   * @returns GetFungibleAssetActivitiesResponse A list of fungible asset metadata
+   */
+  async getFungibleAssetActivities(args: {
+    options?: {
+      pagination?: PaginationArgs;
+      whereCondition?: {
```
Comment (heliuchuan) [inline src/api/fungible_asset.ts:58]: i'm not a big fan of the where condition in options

File: src/internal/fungible_asset.ts [hunk 2]
```diff
@@ -54,3 +59,47 @@ export async function getFungibleAssetMetadata(args: {
...
+      type?: string;
+    };
+  };
+}): Promise<GetFungibleAssetActivitiesResponse> {
+  const { aptosConfig, options } = args;
+
+  const whereCondition: any = {};
+  if (options?.whereCondition?.ownerAddress) {
+    whereCondition.owner_address = {
+      _eq: AccountAddress.fromHexInput(options.whereCondition.ownerAddress).toString(),
```
Comment (heliuchuan) [inline src/internal/fungible_asset.ts:79]: one of the reasons being that options.whereCondition.ownerAddress is a bit clunky.

set it to a variable?

File: src/internal/fungible_asset.ts [hunk 3]
```diff
@@ -54,3 +59,47 @@ export async function getFungibleAssetMetadata(args: {
...
+    whereCondition?: {
+      ownerAddress?: HexInput;
+      assetType?: string;
+      type?: string;
+    };
+  };
+}): Promise<GetFungibleAssetActivitiesResponse> {
+  const { aptosConfig, options } = args;
+
+  const whereCondition: any = {};
```
Comment (heliuchuan) [inline src/internal/fungible_asset.ts:76]: can you type to FungibleAssetActivitiesBoolExp

File: src/api/fungible_asset.ts (hunk 1 above)
Comment (heliuchuan) [reply src/api/fungible_asset.ts:58]: and it would be inconsistent with our other queries I feel

Comment (0xmaayan) [reply src/api/fungible_asset.ts:58]: agreed. also, it makes it hard to support custom user input where condition

File: tests/e2e/api/fungible_asset.test.ts [hunk 4]
```diff
@@ -19,4 +19,23 @@ describe("FungibleAsset", () => {
     expect(data[0]).toHaveProperty("asset_type");
     expect(data[0].asset_type).toEqual(APT_COIN_TYPE);
   });
+
+  test("it should fetch fungible asset activities with correct number and asset type ", async () => {
+    const config = new AptosConfig({ network: Network.LOCAL });
```
Comment (0xmaayan) [inline tests/e2e/api/fungible_asset.test.ts:24]: let's declare all those repated variables (config, aptos, APT_COIN_TYPE, etc) at the top of the file

File: src/internal/fungible_asset.ts [hunk 5]
```diff
@@ -54,3 +59,47 @@ export async function getFungibleAssetMetadata(args: {
 
   return data.fungible_asset_metadata;
 }
+
+export async function getFungibleAssetActivities(args: {
+  aptosConfig: AptosConfig;
+  options?: {
+    pagination?: PaginationArgs;
+    whereCondition?: {
```
Comment (0xmaayan) [inline src/internal/fungible_asset.ts:67]: not completely sure why we need to define what user can filter by? where condition should accept user customized properties, like https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/api/event.ts#L75

File: src/api/fungible_asset.ts [hunk 6]
```diff
@@ -36,4 +41,26 @@ export class FungibleAsset {
...
+   * Queries the fungible asset activities
+   *
+   * This query returns the fungible asset activities.
+   * It can be filtered by owner address, asset type, and type.
+   *
+   * @param options.pagination Where offset and limit can be specified
+   * @param options.whereCondition Where condition to filter the results
+   * @returns GetFungibleAssetActivitiesResponse A list of fungible asset metadata
+   */
+  async getFungibleAssetActivities(args: {
```
Comment (0xmaayan) [inline src/api/fungible_asset.ts:55]:
```suggestion
  async getFungibleAssetActivities(args?: {
```

File: src/api/fungible_asset.ts [hunk 7]
```diff
@@ -36,4 +41,26 @@ export class FungibleAsset {
...
     return getFungibleAssetMetadata({ aptosConfig: this.config, ...args });
   }
+
+  /**
+   * Queries the fungible asset activities
+   *
+   * This query returns the fungible asset activities.
+   * It can be filtered by owner address, asset type, and type.
+   *
+   * @param options.pagination Where offset and limit can be specified
```
Comment (0xmaayan) [inline src/api/fungible_asset.ts:51]: I think we agreed to not add optional arguments to comments

File: src/internal/fungible_asset.ts (hunk 5 above)
Comment (0xjinn) [reply src/internal/fungible_asset.ts:67]: ohh okay, didn't realize you can assign the bool exp like that. 

Will update

**Review by 0xmaayan — DISMISSED** [review]
left comments

File: src/api/fungible_asset.ts (hunk 1 above)
Comment (0xjinn) [reply src/api/fungible_asset.ts]: Updated to use bool exp

File: tests/e2e/api/fungible_asset.test.ts [hunk 8]
```diff
@@ -19,4 +19,23 @@ describe("FungibleAsset", () => {
...
+    const aptos = new Aptos(config);
+    const APT_COIN_TYPE = "0x1::aptos_coin::AptosCoin";
```
[the remark this answers is not in the document]
Comment (0xjinn) [reply tests/e2e/api/fungible_asset.test.ts]: updated.

File: tests/e2e/api/fungible_asset.test.ts (hunk 4 above)
Comment (0xjinn) [reply tests/e2e/api/fungible_asset.test.ts:25]: Good call. I will address this part in the next PR on the stack, so there won't be conflict.

**Comment by 0xjinn** [comment]
Update - Fixed most part from comments. Will address the repeated use of same const variable in the next(last) PR in the stack

**Review by heliuchuan — APPROVED** [review]

**Pull request merged**